data/virt_autotest/setup_dns_service.sh

```diff
@@ -174,48 +229,153 @@ for single_nameserver in ${nameservers_array[@]};do
-	exit 1
+        echo -e "DNS service does not work for at least one vm. Please pay attention !\n" | tee -a ${setup_log_file}
+        exit 1
 else
-	echo -e "SSH Connection to all virtual machines by using DNS names is working now.\n" | tee -a ${setup_log_file}
+	echo -e "DNS service works for al vm guests !\n" | tee -a ${setup_log_file}
+	while [ ${sleep_count} -le ${sleep_time} ];do
+		echo -e "Try $((${sleep_count}/5 + 1)) time(s) ssh-copy-id\n" | tee -a ${setup_log_file}
+        	vmguest_index=0
+		for vmguest in ${vm_guestnames_array[@]};do
+			sshpass -p ${ssh_key_pass} ssh-copy-id -i ${ssh_key_path}/id_rsa.pub -f root@${vmguest}
+			vmguest_sshcopyid_failed[${vmguest_index}]=`echo $?`
+                	vmguest_index=$((${vmguest_index} + 1))
+		done
+		failed_count=0
+                vmguest_index=0
+        	for vmguest in ${vm_guestnames_array[@]};do
+                	if [[ ${vmguest_sshcopyid_failed[${vmguest_index}]} -ne 0 ]];then
+                        	echo -e "SSH COPY ID to ${vmguest} using its dns name failed !\n" | tee -a ${setup_log_file}
+                        	failed_count=$((${failed_count} + 1))
+                	else
+                        	echo -e "SSH COPY ID to ${vmguest} using its dns name succeeded !\n" | tee -a ${setup_log_file}
+                	fi
+                	vmguest_index=$((${vmguest_index} + 1))
+        	done
+        	if [[ ${failed_count} -eq 0 ]];then
+                	echo -e "SSH COPY ID to all virtual machines by using DNS names is working now.\n" | tee -a ${setup_log_file}
+                	break
+        	fi
+	        sleep_count=$((${sleep_count} + 5))
+        	sleep 5s
+	done
 fi
+
+#Verify whether NoPassword ssh connection works for all vm guests.Quit if any failure.
+failed_count=0
+if [[ ${failed_count} -gt 0 ]];then
+        echo -e "Failed to ssh-copy-id for at least one vm. Please pay attention !\n" | tee -a ${setup_log_file}
+        exit 1
+else
+	eval `ssh-agent -t 86400 -s`
+	ssh-add -t 86400 ${ssh_key_path}/id_rsa
+	ssh-add -L
+	echo -e "Succeeded ssh-copy-id to all virtual machines by using DNS names.\n" | tee -a ${setup_log_file}
+	for vmguest in ${vm_guestnames_array[@]};do
+       		echo -e "ssh root@${vmguest} whoami" | tee -a ${setup_log_file}
+		ssh root@${vmguest} whoami
+		if [[ `echo $?` -ne 0 ]];then
+			echo -e "NoPassword SSH connection to ${vmguest} failed !\n" | tee -a ${setup_log_file}
+                	failed_count=$((${failed_count} + 1))
+		else
+			echo -e "NoPassword SSH connection to ${vmguest} succeeded !\n" | tee -a ${setup_log_file}
+        	fi
+	done
+fi
+if [[ ${failed_count} -gt 0 ]];then
+        echo -e "NoPassword SSH connection does not work for at least one vm. Please pay attention !\n" | tee -a ${setup_log_file}
+        exit 1
+else
+        echo -e "NoPassword SSH connection to all virtual machines by using DNS names is working now.\n" | tee -a ${setup_log_file}
+fi
+chmod 777 ${setup_log_file}
```

Comment:
Would you please add some comments describing the purpose(avoid password when visit guest via domain name?) and how you do it at the very beginning of this big section serving this purpose?